### ISSUE TITLE
BrowserslistError: Unknown version 61 of firefox

### ISSUE THREAD

**Issue by laozhu** [jonathantneal/postcss-preset-env#8]
Hello, @jonathantneal 

When I use `postcss-cssnext`, everything is OK.
After I move from `postcss-cssnext` to `postcss-preset-env` plugin, I get this strange error.

![image](https://user-images.githubusercontent.com/1212457/36143622-8fe145a4-10e6-11e8-8ab0-7ac85401ab37.png)

I set browserslist in package.json

```json
{
  "browserslist": [
      "> 1%",
      "last 3 versions",
      "Firefox ESR",
      "not ie <= 8",
      "not op_mini all",
      "not bb 7-10"
  ]
}
```

BTW, is this plugin using `cssnext` as the backend?

**Comment by AlecRust** [comment]
Also getting this with first try of `postcss-preset-env`.

**Comment by jonathantneal** [comment]
I’ve started getting this over the last few days. Something significant has changed in a minor or patch release of Browserslist.

**Comment by ai** [comment]
Sometimes it happens on different Browserslist versions in different tool when one tool pass exact browsers versions to other tool

**Comment by ai** [comment]
How do you use Browserslist?

**Comment by jonathantneal** [comment]
> How do you use Browserslist?

This is happening for React users. I am using [create-react-app](https://github.com/facebook/create-react-app), myself.

---

If you want to know how I use browserslist in postcss-preset-env, I pull a browserslist [from any configuration relative to the css file](https://github.com/jonathantneal/postcss-preset-env/blob/master/index.js#L73-L76):

```js
browserslist(browsers, {
  path: result.root.source && result.root.source.input && result.root.source.input.file
})
```

Then, I compare it to a browserslist [generated by each feature](https://github.com/jonathantneal/postcss-preset-env/blob/master/index.js#L81-L83):

```js
supportedBrowser => browserslist(feature.browsers).some(
  polyfillBrowser => polyfillBrowser === supportedBrowser
)
```

This narrows down my plugins to only the ones required by that first configuration. 😄 

If you want to know how the feature browserslist is generated, I build it using [caniuse-lite](https://github.com/jonathantneal/postcss-preset-env/blob/master/lib/get-unsupported-browsers-by-feature.js).

**Comment by ai** [comment]
What is on `index.bundle.js:160`? Can you repeat error and put a `console.log` there?

**Comment by jonathantneal** [comment]
@ai, `index.bundle.js:160` is where browserslist creates an array from the browsers + versions provided by `caniuse-lite`.

```js
return browserslist(feature.browsers).some(function (polyfillBrowser) {
  return polyfillBrowser === supportedBrowser;
})
```

Does this mean the browser names in caniuse-lite might not match against browsers in browserslist? If so, how would you like me to handle this? @laozhu, still feel free to share your error.

**Comment by ai** [comment]
Yeap. It is a problem. The first idea is to convert `ie 9, ie 10, ie 11` into `ie 9-11` (Browserslist ignores missed version in this case).

The second idea is to add an extra option to Browserslist.

**Comment by jonathantneal** [comment]
Okay, I will look into this. @ai or @laozhu, could you give me an example browserslist query that should break the plugin? I tried the following browserslist, but it did not throw an error for me:

```js
browsers: [
  '> 1%',
  'last 3 versions',
  'Firefox ESR',
  'not ie <= 8',
  'not op_mini all',
  'not bb 7-10'
]
```

**Comment by ai** [comment]
@jonathantneal it is a tricky case - there is no query to break. You must have different `caniuse-lite` versions in Broweerslist and `postcss-preset-env`

**Comment by jonathantneal** [comment]
Thanks, @ai.

1. Is there an option to specify the version of caniuse-lite to be used by browserslist?
2. Could I just use `browserslist` as the dependency and expect `caniuse-lite` from it?

**Comment by ai** [comment]
@jonathantneal 

1. No. It is how npm works.
2. In theory, but it will not 100% bulletproof.

We have better solutions:

1. Send `"ie 9-11"` instead of `["ie 9", "ie 10", "ie 11"]`. Browserslist ignores unknown version in range query.
2. Add `ignoreUnknownVersion` option to Browserslist.

**Comment by jonathantneal** [comment]
@ai

I tried constructing queries as you described, and Browserslist does not like them. An example is:

```
safari 4-TP
```

Where `TP` stands for **Safari Technical Preview**.

**Comment by ai** [comment]
Oh, yeap, non-digit versions :(. OK, I will add `ignoreUnknownVersion` tomorrow.

**Comment by ai** [comment]
Option `ignoreUnknownVersions` was released in Browserslist 3.1.0.
